scripts/gha/utils.py

```diff
@@ -114,10 +179,38 @@ def get_vcpkg_triplet(arch):
     triplet_name.append('linux')
 
   triplet_name = '-'.join(triplet_name)
+  ok = check_vcpkg_triplet(triplet_name, arch, crt_linkage)
+  if not ok:
+    triplet_name = create_vcpkg_triplet(arch, crt_linkage)
+
   print("Using vcpkg triplet: {0}".format(triplet_name))
   return triplet_name
 
 
+def get_vcpkg_triplet_file_path(triplet_name):
+  """Get absolute path to vcpkg triplet configuration file."""
+  triplet_file_path = os.path.join(get_vcpkg_root_path(), 'triplets',
+                                   triplet_name+'.cmake')
+  return triplet_file_path
+
+
+def get_vcpkg_response_file_path(triplet_name):
+  """Get absolute path to vcpkg response file containing packages to install"""
+  response_file_dir_path = os.path.join(os.getcwd(), 'external')
+  response_file_path = os.path.join(response_file_dir_path,
+                       'vcpkg_' + triplet_name + '_response_file.txt')
+  if not os.path.exists(response_file_path):
+    existing_response_file_path = os.path.join(response_file_dir_path,
+                         'vcpkg_x64-linux' + '_response_file.txt')
+    with open(existing_response_file_path, 'r') as existing_file:
+      lines = existing_file.readlines()
+      lines[-1] = triplet_name + '\n'
+      with open(response_file_path, 'w') as response_file:
+        response_file.writelines(lines)
+        print("Created new response file: {0}".format(response_file_path))
```

Comment:
This could use some comments explaining what's going on, as it's very non-obvious (especially why we're using the x64 linux response file as a template).

Reply:
Just added a comment explaining the reason for creating a new response file. x64 linux response file is arbitrarily chosen from the list of files that are shipped in our repo. I was wondering if we should remove all the existing response files from our repo and have one "response_file_template" and modify it around build time. This has the advantage of storing the list of packages to install in one file too. Right now, in order to add/remove a package, you have to modify all response files. 

The disadvantage of removing these response files is users will have to completely depend on our python build scripts to build anything at all. I feel that manually running commands (for advanced users) should be possible, especially for the common cases.